app/src/main/java/com/keylesspalace/tusky/db/TimelineDao.kt

```diff
@@ -1,84 +1,66 @@
+/* Copyright 2021 Tusky Contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
 package com.keylesspalace.tusky.db
 
+import androidx.paging.PagingSource
 import androidx.room.Dao
 import androidx.room.Insert
-import androidx.room.OnConflictStrategy.IGNORE
 import androidx.room.OnConflictStrategy.REPLACE
 import androidx.room.Query
-import androidx.room.Transaction
-import io.reactivex.rxjava3.core.Single
 
 @Dao
 abstract class TimelineDao {
 
     @Insert(onConflict = REPLACE)
-    abstract fun insertAccount(timelineAccountEntity: TimelineAccountEntity): Long
+    abstract suspend fun insertAccount(timelineAccountEntity: TimelineAccountEntity): Long
 
     @Insert(onConflict = REPLACE)
-    abstract fun insertStatus(timelineAccountEntity: TimelineStatusEntity): Long
-
-    @Insert(onConflict = IGNORE)
-    abstract fun insertStatusIfNotThere(timelineAccountEntity: TimelineStatusEntity): Long
+    abstract suspend fun insertStatus(timelineStatusEntity: TimelineStatusEntity): Long
 
     @Query(
         """
 SELECT s.serverId, s.url, s.timelineUserId,
 s.authorServerId, s.inReplyToId, s.inReplyToAccountId, s.createdAt,
 s.emojis, s.reblogsCount, s.favouritesCount, s.reblogged, s.favourited, s.bookmarked, s.sensitive,
 s.spoilerText, s.visibility, s.mentions, s.application, s.reblogServerId,s.reblogAccountId,
-s.content, s.attachments, s.poll, s.muted,
+s.content, s.attachments, s.poll, s.muted, s.expanded, s.contentShowing, s.contentCollapsed, s.pinned,
```

Comment:
I thought you preserve these fields in memory anyway?

Reply:
well yes, but the UI does still only get updated through the db